### PR TITLE
179376124 Page Navigation and Copy Part 3

### PR DESCRIPTION
**Pull request by knowuh**
This PR fixes several bugs with the initial Navigation and Page copy work.

- Fixes bugs with Move Item when specifying a page
- Fixes bugs with Move Section when specifying a page
- Consolidates duplicate concerns of moving things 
- Copy Page will now honor destination (where to copy to)

### CODE REVIEW

**Review by knowuh — COMMENTED** [review]
I can't delete my comment.

File: lara-typescript/src/section-authoring/api/api-types.ts [hunk 1]
```diff
@@ -160,14 +160,19 @@ export interface IPage {
    */
   sections: ISection[];
 
+  /**
+   * Sections on this page:
```
Comment (emcelroy) [inline lara-typescript/src/section-authoring/api/api-types.ts:164]: Should this comment be changed?

File: app/controllers/api/v1/interactive_pages_controller.rb [hunk 2]
```diff
@@ -76,7 +86,7 @@ def set_sections
     # Remove deleted sections:
     old_sections.each do |section|
       unless (new_section_ids.include?(section.id.to_s))
-        section.delete
+        section.update_attribute(:interactive_page_id, nil)
```
Comment (emcelroy) [inline app/controllers/api/v1/interactive_pages_controller.rb:89]: Sorry but I don't quite understand this change. I understand that it's setting the section's interactive_page_id value to nil and that that must negate its association with the page. But I don't understand why that's an improvement over `section.delete`. Do the deleted sections need to kept around for some reason even though they're no longer associated with the page? Is this related to the handling of moving a section between pages? I'm guessing so, but it's not quite clear to me.

**Review by emcelroy — APPROVED** [review]
I think this looks good. There's one part I had a question about, but I'm probably just not fully understanding what's being done rather than it being a problem.

File: app/controllers/api/v1/interactive_pages_controller.rb (hunk 2 above)
Comment (knowuh) [reply app/controllers/api/v1/interactive_pages_controller.rb:89]: When we 'move' a section, we just remove its association from the page where it is now, and add a new association to its destination. We update each page separately. 

Before this change, we would lose sections when moving them in LARA sometimes.

**Comment by knowuh** [comment]
Thanks @emcelroy

**Pull request merged**